pkg/envoy/lds/response.go

```diff
@@ -55,13 +65,23 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_disco
 	inboundListener := newInboundListener()
 	if meshFilterChain, err := getInboundInMeshFilterChain(proxyServiceName, cfg); err != nil {
 		log.Error().Err(err).Msgf("Error making in-mesh filter chain for proxy %s", proxy.GetCommonName())
-	} else if meshFilterChain != nil {
+	} else {
+		if !cfg.IsPermissiveTrafficPolicyMode() {
+			// Apply RBAC policies on the inbound filters based on configured policies
+			rbacFilter, err := lb.buildRBACFilter()
+			if err != nil {
+				log.Error().Err(err).Msgf("Error applying RBAC filter for service %s", proxyServiceName)
+				return nil, err
+			}
+			// RBAC filter should be the very first filter in the filter chain
```

Comment:
Do we have a test to ensure that this is always the case (RBAC... be the very first filter)?  I can imagine it easy to break this requirement the next time around we add a filter and it seems very important.

Reply:
We don't have a test for the entire LDS response, which is much needed and a work in progress. i'd prefer to tackle that in a separate PR since the code and tests need more refactoring to be able to use mocks, listenerBuilder struct etc.

Reply:
I created an issue for this missing test in general: https://github.com/openservicemesh/osm/issues/2056

With this PR, we are assured that the RBAC filter will be the very first filter because of how the filter is being appended to the existing slice of filters (internal golang behavior).